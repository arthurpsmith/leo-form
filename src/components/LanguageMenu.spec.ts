import { createLocalVue, mount } from '@vue/test-utils'
import LanguageMenu from '@/components/LanguageMenu.vue'
import Vuetify from 'vuetify'

describe('LanguageMenu.vue', () => {
  const localVue = createLocalVue()
  let vuetify: Vuetify
  function mountFunction (options = {}) {
    options = {
      data: {
        locales: [
          {
            locale: 'en',
            title: 'English'
          },
          {
            locale: 'fr',
            title: 'Français'
          },
          {
            locale: 'es',
            title: 'Español'
          }
        ]
      },
      ...options
    }
    return mount(LanguageMenu, {
      localVue,
      vuetify,
      ...options
    })
  }
  beforeEach(() => {
    vuetify = new Vuetify()
  })
  it('should work', () => {
    const wrapper = mountFunction()

    expect(wrapper.html()).toMatchSnapshot()
  })
  it('Has the right icon', () => {
    const wrapper = mountFunction()
    expect(wrapper.find('i.v-icon').exists()).toBe(true)
  })
  it('Uses correct translation key for button text', () => {
    const wrapper = mountFunction()
    const comp = wrapper.find('.v-btn__content')
    expect(comp.text()).toEqual('$vuetify.language_menu_button')
  })
  it('Has a menu entry for each locale', async () => {
    const wrapper = mountFunction()
    await wrapper.find('button').trigger('click')
    const menuItems = wrapper.findAll('.v-list-item')
    expect(menuItems).toHaveLength(3)
  })
  // Probably a better candidate for an e2e test
  // it('Calls the changeLocale method with the new locale on menu item click', async () => {
  //   const wrapper = mountFunction()
  //   await wrapper.find('button').trigger('click')
  //   const menuItems = wrapper.findAll('.v-list-item')
  //   const newLocateItem = menuItems.at(2)
  //   await newLocateItem.trigger('click')
  //   expect(spy).toBeCalledTimes(1)
  //   expect(menuItems).toHaveLength(3)
  // })
})
