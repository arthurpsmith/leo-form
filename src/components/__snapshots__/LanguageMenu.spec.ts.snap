// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LanguageMenu.vue should work 1`] = `
<div class="v-menu"><button type="button" class="v-btn v-btn--outlined theme--light v-size--default primary--text" role="button" aria-haspopup="true" aria-expanded="false"><span class="v-btn__content"><i aria-hidden="true" class="v-icon notranslate v-icon--left mdi mdi-translate theme--light"></i>
      $vuetify.language_menu_button
    </span></button>
  <!---->
</div>
`;
